sdk-java/src/main/java/ly/count/sdk/java/internal/ModuleBackendMode.java

```diff
@@ -0,0 +1,434 @@
+package ly.count.sdk.java.internal;
+
+import ly.count.sdk.java.Countly;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.*;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+public class ModuleBackendMode extends ModuleBase {
+
+    protected static final Log.Module L = Log.module("BackendMode");
+    protected InternalConfig internalConfig = null;
+    protected CtxCore ctx = null;
+
+    //disabled is set when a empty module is created
+    //in instances when the rating feature was not enabled
+    //when a module is disabled, developer facing functions do nothing
+    protected boolean disabledModule = false;
+
+    private int eventQSize = 0;
+    protected boolean defferUpload = false;
+    private final Queue<Request> requestQ = new LinkedList<>();
+    private final Map<String, JSONArray> eventQueues = new HashMap<>();
+
+    private Tasks tasks;
```

Comment:
This should not exist

Reply:
We are using it to process request queue.

Reply:
Yes, the processing should be reworked